actionview/lib/action_view/helpers/url_helper.rb

```diff
@@ -589,10 +589,16 @@ def link_to_remote_options?(options)
         end
 
         def add_method_to_attributes!(html_options, method)
-          if method && method.to_s.downcase != "get".freeze && html_options["rel".freeze] !~ /nofollow/
-            html_options["rel".freeze] = "#{html_options["rel".freeze]} nofollow".lstrip
+          @_method_downcase_cache ||= {}
+          downcased_method = @_method_downcase_cache[method] ||= method.to_s.downcase
+          if downcased_method && downcased_method != "get" && html_options["rel"] !~ /nofollow/
+            if html_options["rel"].present?
+              html_options["rel"] = "#{html_options["rel"]} nofollow"
+            else
+              html_options["rel"] = "nofollow"
```

Comment:
Thoughts on this change separate from the cache change? It's also a performance improvement for the case where there is no `html_options["rel"]` value

Reply:
Just double checked this and it prevents an allocation of `""` for each `link_to` call that does not pass in a `"rel"` option.

Reply:
Moved this to https://github.com/rails/rails/pull/30997